isis/src/base/objs/ProcessImport/ProcessImport.cpp

```diff
@@ -1199,21 +1199,26 @@ namespace Isis {
     }
   }
 
-
   /**
-   * Create the output file. Note that all the appropiate calls to at least
-   * SetDimensions and SetPixelType should be made prior to calling this method.
+   * Create the output file. This method will use the cube attribute object
+   * passed in to populate the pixel type and pass the desired samples, lines,
+   * and band numbers into the process setOutputCube method.
    *
    * @param parameter The parameter name that holds the output file name.
    *
+   * @param att The CubeAttributeOutput object with defined attributes
+   *
+   * @param ns The number of samples in the output cube
+   *
+   * @param nl The number of lines in the output cube
+   *
+   * @param nb The number of bands in the output cube
+   *
    * @return @b Isis::Cube Output cube.
    *
    * @throws Isis::iException::Message "Unsupported pixel type."
    */
-  Isis::Cube *ProcessImport::SetOutputCube(const QString &parameter) {
-    CubeAttributeOutput &att =
-      Application::GetUserInterface().GetOutputAttribute(parameter);
-
+  Isis::Cube *ProcessImport::SetOutputCube(const QString &parameter, CubeAttributeOutput &att, const int ns, const int nl, const int nb) {
```

Comment:
Since this call is being passed the user interface parameter name, I don't see the reason to pass in the CubeAttribute. It is part of what the user typed. Also, the original code parsed the value of the parameter and pulled the attributes off in order to call the parents member. Seems reasonable to continue doing this and not make the application programmer do this.

Reply:
No longer necessary as the change to ProcessImport was unnecessary